core/src/main/resources/xml/xsl/adapter2mermaid.xsl

```diff
@@ -0,0 +1,552 @@
+		<xsl:value-of select="tokenize(@className, '\.')[last()]"/>
+		<xsl:text>])</xsl:text>
+		<xsl:text>&#10;</xsl:text>
+	</xsl:template>
+
+	<xsl:template match="outputWrapper" mode="convertElements">
+		<xsl:if test="@errorHandling = 'true'">
+			<xsl:text>	style </xsl:text>
+			<xsl:value-of select="@elementID"/>
+			<xsl:text> stroke-dasharray: 4 4</xsl:text>
+			<xsl:text>&#10;</xsl:text>
+		</xsl:if>
+		<xsl:text>	</xsl:text>
+		<xsl:value-of select="@elementID"/>
+		<xsl:text disable-output-escaping="yes">([OutputWrapper&lt;br/></xsl:text>
+		<xsl:value-of select="tokenize(@className, '\.')[last()]"/>
+		<xsl:text>])</xsl:text>
+		<xsl:text>&#10;</xsl:text>
+	</xsl:template>
+
+	<xsl:template match="pipe" mode="convertElements">
+		<xsl:variable name="isSwitch" select="@className='nl.nn.adapterframework.pipes.XmlSwitch'
+											or@className='nl.nn.adapterframework.pipes.CompareIntegerPipe'
+											or@className='nl.nn.adapterframework.pipes.CompareStringPipe'
+											or@className='nl.nn.adapterframework.pipes.FilenameSwitch'
+											or@className='nl.nn.adapterframework.pipes.XmlIf'"/>
+		<xsl:if test="@errorHandling = 'true'">
+			<xsl:text>	style </xsl:text>
+			<xsl:value-of select="@elementID"/>
+			<xsl:text> stroke-dasharray: 4 4</xsl:text>
+			<xsl:text>&#10;</xsl:text>
+		</xsl:if>
+		<xsl:text>	</xsl:text>
+		<xsl:value-of select="@elementID"/>
+		<xsl:choose>
+			<xsl:when test="$isSwitch">{</xsl:when>
+			<xsl:otherwise>(</xsl:otherwise>
+		</xsl:choose>
+		<xsl:value-of select="@name"/>
+		<xsl:text disable-output-escaping="yes">&lt;br/></xsl:text>
+		<xsl:choose>
+			<xsl:when test="@className = 'nl.nn.adapterframework.pipes.GenericMessageSendingPipe'">
+				<xsl:value-of select="tokenize(sender/@className, '\.')[last()]"/>
+			</xsl:when>
+			<xsl:otherwise>
+				<xsl:value-of select="tokenize(@className, '\.')[last()]"/>
+			</xsl:otherwise>
+		</xsl:choose>
+		<xsl:choose>
+			<xsl:when test="$isSwitch">}</xsl:when>
+			<xsl:otherwise>)</xsl:otherwise>
+		</xsl:choose>
+		<xsl:text>&#10;</xsl:text>
+	</xsl:template>
+
+	<xsl:template match="exit" mode="convertElements">
+		<xsl:text>	</xsl:text>
+		<xsl:value-of select="@elementID"/>
+		<xsl:text>{{</xsl:text>
+		<xsl:value-of select="@state"/>
+		<xsl:if test="@code">
+			<xsl:text disable-output-escaping="yes">&lt;br/></xsl:text>
+			<xsl:value-of select="@code"/>
+		</xsl:if>
+		<xsl:text>}}</xsl:text>
+		<xsl:text>&#10;</xsl:text>
+	</xsl:template>
+
+	<xsl:template match="forward" mode="convertForwards">
+		<xsl:text>	</xsl:text>
+		<xsl:value-of select="parent::*/@elementID"/>
+		<xsl:value-of select="if (@errorHandling = 'true') then (' -. ') else (' --> |')" disable-output-escaping="yes"/>
+		<xsl:choose>
+			<xsl:when test="exists(@customText)">
+				<xsl:value-of select="@name" disable-output-escaping="yes"/>
+				<xsl:text disable-output-escaping="yes">&lt;br/></xsl:text>
+				<xsl:value-of select="@customText" disable-output-escaping="yes"/>
```

Comment:
Je hebt altijd de regel met `@name`, dus die altijd doen en dan wat specifiek voor customText is in en xsl:if doen die daarna komt